utils/generate-catalog.sh

```diff
@@ -34,7 +35,8 @@ ${OPM} init limitador-operator --default-channel=${CHANNELS} --output yaml >> ${
 cat ${TMP_DIR}/limitador-operator-bundle.yaml >> ${CATALOG_FILE}
 # Add a channel entry for the bundle
 V=`${YQ} eval '.name' ${TMP_DIR}/limitador-operator-bundle.yaml` \
+REPLACES=limitador-operator.v${REPLACES_VERSION} \
 CHANNELS=${CHANNELS} \
-    ${YQ} eval '(.entries[0].name = strenv(V)) | (.name = strenv(CHANNELS))' ${CATALOG_BASEDIR}/limitador-operator-channel-entry.yaml >> ${CATALOG_FILE}
+    ${YQ} eval '(.entries[0].name = strenv(V)) | (.entries[0].replaces = strenv(REPLACES)) | (.name = strenv(CHANNELS))' ${CATALOG_BASEDIR}/limitador-operator-channel-entry.yaml >> ${CATALOG_FILE}
```

Comment:
I would create temp file and apply all changes (replaces conditionally)

```
# copy to temp
cp ${CATALOG_BASEDIR}/limitador-operator-channel-entry.yaml /tmp.dir/limitador-operator-channel-entry.yaml 
# apply entry name
V="${YQ} eval '.name' ${TMP_DIR}/limitador-operator-bundle.yaml" $(YQ) eval '.entries[0].name = strenv(V)' -i /tmp.dir/limitador-operator-channel-entry.yaml 
# apply replaces (optionally)
[[ ! -z "$REPLACES" ]] && V="$REPLACES" $(YQ) eval '.spec.replaces = strenv(V)' -i /tmp.dir/limitador-operator-channel-entry.yaml 
# apply channels (will always have value)
V="$CHANNELS" $(YQ) eval '.name = strenv(V)' -i /tmp.dir/limitador-operator-channel-entry.yaml 

# append to catalog
yq eval /tmp.dir/limitador-operator-channel-entry.yaml >> ${CATALOG_FILE}
```

Reply:
I agree, however since it's the same in the kuadrant operator, I'd say we could do it right when we migrate to a new repo "kuadrant-pipelines" so we can make the change just once. What do you think?

Reply:
ok, no problem.